ampd/src/main.rs

```diff
@@ -150,3 +168,27 @@ enum Error {
     #[error("fatal failure")]
     Fatal,
 }
+
+async fn worker_address(args: &Args) -> ExitCode {
+    let config = init_config(&args.config);
+    let state_path = expand_home_dir(&args.state);
+
+    let state = match state::load(&state_path).change_context(Error::Fatal) {
+        Ok(state) => state,
+        Err(report) => {
+            eprintln!("{report:?}");
+            return ExitCode::FAILURE;
```

Comment:
reuse the setup for error logging and exit codes from the main() function, i.e. return the report instead of an error code and don't print the error here

Reply:
this goes for all spots this function does this